### PR TITLE
Obfuscated id validation implement

### PR DESCRIPTION
**Pull request by zkzhao333**
Implement obfuscated id validation; Added copyright comments.

### CODE REVIEW

File: GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/PurchaseController.cs [hunk 1]
```diff
@@ -210,7 +227,7 @@ public PurchaseProcessingResult ProcessPurchase(PurchaseEventArgs args)
 #endif
     }
 
-    private static bool ClientSideReceiptValidation(string receipt)
+    private static bool ClientSideReceiptValidation(string unityIAPReceipt)
```
Comment (Qing451800) [inline GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/PurchaseController.cs]: unityIapReceipt

File: GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Utils/TrivialKartClientUtil.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Security.Cryptography;
+using System.Text;
+using UnityEngine;
+
+public class TrivialKartClientUtil
+{
+    //Creat and return obfuscated account id based on the sha256 hash result of device unique id
```
Comment (Qing451800) [inline GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Utils/TrivialKartClientUtil.cs]: // Create

Comment (Qing451800) [reply GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Utils/TrivialKartClientUtil.cs]: Also mention that we use device id as a simple example for simplicity.

File: GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Utils/TrivialKartClientUtil.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    public static string GetObfuscatedAccountId()
+    {
+        // TODO: Added online version account id.
```
Comment (Qing451800) [inline GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Utils/TrivialKartClientUtil.cs:25]: Sync with Rafael about this.

**Review by Qing451800 — APPROVED** [review]

File: GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Utils/TrivialKartClientUtil.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        byte[] bytes = Encoding.Unicode.GetBytes(SystemInfo.deviceUniqueIdentifier);
+        SHA256CryptoServiceProvider sha256 = new SHA256CryptoServiceProvider();
+        byte[] hashBytes = sha256.ComputeHash(bytes);
+        StringBuilder hashStringBuilder = new StringBuilder();
+
+        foreach (byte b in hashBytes)
+        {
+            hashStringBuilder.AppendFormat("{0:X2}", b);
```
Comment (Qing451800) [inline GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Utils/TrivialKartClientUtil.cs:33]: Is this a standard way?

Comment (zkzhao333) [reply GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Utils/TrivialKartClientUtil.cs:33]: I am not sure, but I searched online. Lots of people are doing in this way.

**Pull request merged**